docs/getting-started.md

```diff
@@ -3,33 +3,111 @@ layout: docs
 
-These steps relate to the hosted version of the platform; if you wish to host it yourself on your infrastructure, check the [GitHub repository](https://github.com/eduardoboucas/staticman).
+## Step 1. Git provider authentication
 
-## Step 1. Add Staticman to your repository
+Staticman currently supports the following git providers:
 
-I will need push access to the repositories you want me to commit the files to. In GitHub, go to your repository **Settings** page, navigate to the **Collaborators** tab and add the username **staticmanapp**.
+- GitHub
+- GitLab
 
-![Step 1](/assets/images/get-started/step1.png)
+In order to use Staticman, the repository for your static site must be hosted on one of these providers. The Staticman service needs to authenticate with the git provider to commit files and handle pull requests. This step will walk you through obtaining the necessary credentials.
 
-At this point, the invitation will be pending. In order for me to accept it, you need to open:
+### **If using GitLab to host the static site repo:**
 
-`https://api.staticman.net/v2/connect/{your GitHub username}/{your repository name}`
+[Create a personal access token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#creating-a-personal-access-token) with the following scopes
 
-If all goes well, you should receive a message saying **OK!** and we're good to go.
+- `read_repository`: Necessary to read the Staticman site config
+- `write_repository`: Necessary to merge pull requests
 
-## Step 2. Create a configuration file
+### **If using GitHub to host the static site repo:**
 
-I will look for a `staticman.yml` file in the root of the repository, where various configuration parameters will be defined.
+#### **Option 1. Authenticate as a GitHub application**
 
-You can use the [sample config file](https://github.com/eduardoboucas/staticman/blob/master/staticman.sample.yml) as a starting point and check the [available configuration parameters](/docs/configuration) for more information.
+This is the recommended way to authenticate with GitHub. This method will give the most control over what Staticman can and can't access.
 
-## Step 3. Hook up your forms
+1. <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/creating-a-github-app" class="cta">Create a new GitHub application</a>. Ensure you use the following:
+  - Homepage: `"https://staticman.net/"`
+  - Webhook URL: `"{STATICMAN_BASE_URL}/v1/webhook"` - e.x. `"https://mystaticmaninstance.herokuapp.com/v1/webhook"`
+  - Contents: `Read & Write` - Necessary to read the Staticman site config
+  - Pull Requests: `Read & Write` - Necessary to merge pull requests
+  - Subscribe to `Pull request` events
+1. Generate a private key for the app and note it down along with your app ID.
+1. Install the app on your GitHub account. You can limit its access to only the repo from which you host your static site
```

Comment:
I use a setup of two repos, one that contains source and the other which contains the deployment. In this case, we are probably referring to the source version, since it will contain `_data`. Maybe adding a `source version of static site` will make it more specific.